src/core/ext/filters/client_channel/resolver/xds/xds_resolver.cc

```diff
@@ -29,6 +29,7 @@
 #include "src/core/lib/channel/channel_args.h"
 #include "src/core/lib/iomgr/closure.h"
 #include "src/core/lib/iomgr/exec_ctx.h"
+#include "src/core/lib/security/authorization/evaluate_args.h"
```

Comment:
I don't think the xds resolver should directly depend on the authz library.  Instead, I suggest moving `GetMetadataValue()` into src/core/lib/transport/metadata_batch.{h,cc}, where it can be part of the `grpc_metadata_batch` API.  Then we can use that function both here and in `EvaluateArgs`.

A few details to get right:
- For consistency with the rest of the `grpc_metadata_batch` API, we should call the function something like `grpc_metadata_batch_get_value()`, and we should make `initial_metadata` the first parameter.
- It looks like we don't have any existing unit tests for the `grpc_metadata_batch` API, but let's add one for this method.  It can be added to test/core/transport/metadata_test.cc.
- Please make sure to document the method's behavior in metadata_batch.h.

Reply:
Sorry for the delay in resolving comments. 
I added the method to grpc_metadata_batch as per your recommendation.